server/__tests__/signin.spec.js

```diff
@@ -0,0 +1,117 @@
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../../app';
+import baseUrl from './utils/baseUrl.utils';
+import mock from './utils/mock';
+
+chai.use(chaiHttp);
+
+describe('User Sign In', () => {
+  it('should successfully sign in an existing user', async () => {
+    const res = await chai
+      .request(app)
+      .post(`${baseUrl}/auth/signin`)
+      .send({
+        ...mock.signin
+      });
+    expect(res.status).to.equal(200);
+    expect(res.body.message).to.equal('Successfully signed in');
+    expect(res.body.data[0].user.email).to.equal(mock.signin.email);
+  });
+
+  it('should not signin a non-existing email', async () => {
+    const res = await chai
+      .request(app)
+      .post(`${baseUrl}/auth/signin`)
+      .send({
+        ...mock.signin,
+        email: 'holly@holly.com',
+      });
+    expect(res.status).to.equal(404);
+    expect(res.body.errors.email).to.equal('Email or password is incorrect');
+  });
+
+  it('should not signin a non-existing username', async () => {
+    const res = await chai
+      .request(app)
+      .post(`${baseUrl}/auth/signin`)
+      .send({
+        username: 'holly',
+        password: 'Mother2019'
+      });
+    expect(res.status).to.equal(404);
+    expect(res.body.errors.username).to.equal('Username or password is incorrect');
+  });
+
+  it('should not signin with wrong password', async () => {
+    const res = await chai
+      .request(app)
+      .post(`${baseUrl}/auth/signin`)
+      .send({
+        ...mock.signin,
+        password: 'Father2019'
+      });
+    expect(res.status).to.equal(400);
+    expect(res.body.errors.password).to.equal('Password is incorrect');
```

Comment:
I think this poses a security risk. Getting this message, a hacker would hit a jackpot knowing that the provided `username/email` was correct. You can use a generic message like `invalid credentials supplied` as is the common practice.

Reply:
Thanks...will make the changes